warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -8754,7 +8781,17 @@ public static GTSEncoder shrinkTo(GTSEncoder encoder, int newsize) throws IOExce
     if (0 == lastchunk) {
       lastchunkid = newestChunk;
     }
-
+    
+    //
+    // If chunkcount is Integer.MAX_VALUE, we do not care about the number of chunks,
+    // so if keepempty is false we only need to iterate over the chunks from newestChunk to
+    // oldestChunk
+    //
+    
+    if (!keepempty && zeroChunkCount) {
+      lastchunkid = newestChunk;
+    }
```

Comment:
You must add `if (lastchunkid > newestChunk)`.

Also, the same idea as above comment can apply here: drop `zeroChunkCount` and update `chunkcount` with `chunkcount -= (lastchunkid - newestChunk) / chunkwidth`